README.md

```diff
@@ -2,36 +2,236 @@
 
 [![Build Status](https://travis-ci.org/pglbutt/noodles.svg?branch=master)](https://travis-ci.org/pglbutt/noodles)
 
-The only command line client worthy of pglbutts
+Noodles is a collection of command line tools designed to make your life better.
 
-## How
+# spag
+
+spag is a __command line REST client__. It aims to make developing, testing, and using REST APIs as easy as possible. spag lets you:
+
+- Switch between multiple environmenst with variables for use with multiple APIs.
+- Define parameterized requests in YAML files.
+- Grab data from the previous response for use in future requests.
+- Examine a detailed history of requests.
+
+
+## TODO BEFORE 0.0.1 RELEASE
+
+* Use environment variables in requests, like `spag get /things/@env.id`
+* Use previous request data in template `--with` clauses. `spag request get_thing --with id=@id`
+* More things
+
+## Install
+
+#### Recommended
+
+```bash
+pip install spag
+```
+
+#### From source
 
 ```bash
 git clone https://github.com/pglbutt/noodles.git
 cd noodles
-virtualenv env
-source env/bin/activate
-pip install -r requirements.txt
-pip install --editable .
-spag set http://localhost:5000
-spag show
-http://localhost:5000
-spag get /
+pip install .
+```
+
+## Basics
+
```

Comment:
Can we put some words here for an overview of this section? Like maybe:

- "Spag can function as a replacement for curl for many kinds of requests." (spag doesn't support basic auth. does https even work? I guess it should, but we don't have an equivalent to `-k` in curl)
- "Spag separates the base api endpoint from resource paths to make it easy to work with different deployments of the same api."

Or whatever.